libr/include/r_anal.h

```diff
@@ -215,6 +215,33 @@ enum {
 	R_ANAL_DIFF_TYPE_UNMATCH = 'u'
 };
 
+typedef struct enum_case {
+	const char *name;
+	const int val;
+} RAnalEnumCase;
+
+typedef enum {
+	R_ANAL_BASE_TYPE_KIND_STRUCT,
+	R_ANAL_BASE_TYPE_KIND_UNION,
+	R_ANAL_BASE_TYPE_KIND_ENUM,
+} RAnalBaseTypeKind;
+
+typedef struct r_anal_base_type_struct {
+	RVector *members; // describe the struct in some way
+} RAnalBaseTypeStruct;
+
+typedef struct r_anal_base_type_enum {
+	RVector/*<RAnalEnumCase*/ *cases; // list of all the enum cases
+} RAnalBaseTypeEnum;
+
+typedef struct r_anal_base_type {
+	RAnalBaseTypeKind kind;
+	union {
+		RAnalBaseTypeStruct ztruct;
+		RAnalBaseTypeEnum inum;
```

Comment:
Those zstruct and inum names were more of a joke tbh, we should probably find better names for them. Maybe `struct_data` and `enum_data` or something.

Reply:
Ye, I've figured, but I didn't come up with a better name so I've kept that. Changed it to what you've suggested now.